x-pack/plugins/security_solution/server/lib/risk_engine/utils/create_datastream.ts

```diff
@@ -0,0 +1,206 @@
+      `Ignored PUT mappings for alias ${alias}; error generating simulated mappings: ${err.message}`
+    );
+    return;
+  }
+
+  const simulatedMapping = get(simulatedIndexMapping, ['template', 'mappings']);
+
+  if (simulatedMapping == null) {
+    logger.error(`Ignored PUT mappings for alias ${alias}; simulated mappings were empty`);
+    return;
+  }
+
+  try {
+    await retryTransientEsErrors(
+      () => esClient.indices.putMapping({ index, body: simulatedMapping }),
+      { logger }
+    );
+
+    return;
+  } catch (err) {
+    logger.error(`Failed to PUT mapping for alias ${alias}: ${err.message}`);
+    throw err;
+  }
+};
+/**
+ * Updates the underlying mapping for any existing concrete indices
+ */
+const updateIndexMappings = async ({
+  logger,
+  esClient,
+  totalFieldsLimit,
+  concreteIndices,
+}: UpdateIndexMappingsOpts) => {
+  logger.debug(`Updating underlying mappings for ${concreteIndices.length} indices.`);
+
+  // Update total field limit setting of found indices
+  // Other index setting changes are not updated at this time
+  await Promise.all(
+    concreteIndices.map((index) =>
+      updateTotalFieldLimitSetting({ logger, esClient, totalFieldsLimit, concreteIndexInfo: index })
+    )
+  );
+
+  // Update mappings of the found indices.
+  await Promise.all(
+    concreteIndices.map((index) =>
+      updateUnderlyingMapping({ logger, esClient, totalFieldsLimit, concreteIndexInfo: index })
+    )
+  );
+};
+
+interface CreateConcreteWriteIndexOpts {
+  logger: Logger;
+  esClient: ElasticsearchClient;
+  totalFieldsLimit: number;
+  indexPatterns: IIndexPatternString;
+}
+/**
+ * Create a data stream
+ */
+export const createDataStream = async ({
+  logger,
+  esClient,
+  indexPatterns,
+  totalFieldsLimit,
+}: CreateConcreteWriteIndexOpts) => {
+  logger.info(`Creating data stream - ${indexPatterns.alias}`);
+
+  // check if a datastream already exists
+  let dataStreams: ConcreteIndexInfo[] = [];
+  try {
+    // Specify both the index pattern for the backing indices and their aliases
+    // The alias prevents the request from finding other namespaces that could match the -* pattern
+    const response = await retryTransientEsErrors(
+      () => esClient.indices.getDataStream({ name: indexPatterns.alias, expand_wildcards: 'all' }),
+      { logger }
+    );
+
+    dataStreams = response.data_streams.map((dataStream) => ({
+      index: dataStream.name,
+      alias: dataStream.name,
+      isWriteIndex: true,
+    }));
+
+    logger.debug(
+      `Found ${dataStreams.length} concrete indices for ${indexPatterns.alias} - ${JSON.stringify(
+        dataStreams
+      )}`
+    );
+  } catch (error) {
+    // 404 is expected if no datastream have been created
+    if (error.statusCode !== 404) {
+      logger.error(
+        `Error fetching concrete indices for ${indexPatterns.alias} pattern - ${error.message}`
+      );
+      throw error;
+    }
+  }
+
+  const isDataStreamsExist = dataStreams.length > 0;
```

Comment:
super duper nit: omitting the `is` here makes this more legible; we already have a verb in `Exist` so the `is` is redundant.
```suggestion
  const dataStreamsExist = dataStreams.length > 0;
```